From 09ea9a3aa9643a5d438f1600687d0b3b63f0f388 Mon Sep 17 00:00:00 2001
From: Stefan Csomor <csomor@advancedconcepts.ch>
Date: Wed, 22 Oct 2025 22:02:49 +0200
Subject: [PATCH 01/48] OSX extend private.h usage for __DARWIN__ only builds

eg when building wxGTK on macOS, expose non GUI native utilities and types
---
 include/wx/defs.h        | 4 ++--
 include/wx/osx/private.h | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/include/wx/defs.h b/include/wx/defs.h
index 19ec0b67d3..6cfbf5bebd 100644
--- a/include/wx/defs.h
+++ b/include/wx/defs.h
@@ -2755,7 +2755,7 @@ DECLARE_WXCOCOA_OBJC_CLASS(NSMutableArray);
 DECLARE_WXCOCOA_OBJC_CLASS(NSString);
 DECLARE_WXCOCOA_OBJC_CLASS(NSObject);
 
-#if wxOSX_USE_COCOA
+#ifdef __WXOSX_COCOA__
 
 DECLARE_WXCOCOA_OBJC_CLASS(NSApplication);
 DECLARE_WXCOCOA_OBJC_CLASS(NSBitmapImageRep);
@@ -2811,7 +2811,7 @@ typedef WX_NSOpenGLContext WXGLContext;
 typedef WX_NSPasteboard OSXPasteboard;
 typedef WX_WKWebView OSXWebViewPtr;
 
-#elif wxOSX_USE_IPHONE
+#elif defined(__WXOSX_IPHONE__)
 
 DECLARE_WXCOCOA_OBJC_CLASS(UIColor);
 DECLARE_WXCOCOA_OBJC_CLASS(UIMenu);
diff --git a/include/wx/osx/private.h b/include/wx/osx/private.h
index 23fd0025fb..30a3a84f7c 100644
--- a/include/wx/osx/private.h
+++ b/include/wx/osx/private.h
@@ -3,9 +3,9 @@
 
 #include "wx/osx/core/private.h"
 
-#if wxOSX_USE_IPHONE
+#ifdef __WXOSX_IPHONE__
     #include "wx/osx/iphone/private.h"
-#elif wxOSX_USE_COCOA
+#elif defined(__WXOSX_COCOA__)
     #include "wx/osx/cocoa/private.h"
 #elif wxUSE_GUI
     #error "Must include wx/defs.h first"
-- 
2.24.3 (Apple Git-128)

